hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/exec/processors/CustomContractCreationProcessorTest.java

```diff
@@ -0,0 +1,157 @@
+        given(worldUpdater.getOrCreate(EIP_1014_ADDRESS)).willReturn(contract);
+        given(frame.getValue()).willReturn(WEI_VALUE);
+
+        subject.start(frame, tracer);
+
+        verify(worldUpdater)
+                .tryTransferFromContract(NON_SYSTEM_LONG_ZERO_ADDRESS, EIP_1014_ADDRESS, WEI_VALUE.toLong(), false);
+        verify(mutableContract).setNonce(INITIAL_CONTRACT_NONCE);
+        verify(frame).setState(MessageFrame.State.CODE_EXECUTING);
+    }
+
+    @Test
+    void haltsOnFailedTransfer() {
+        given(frame.getSenderAddress()).willReturn(NON_SYSTEM_LONG_ZERO_ADDRESS);
+        given(frame.getContractAddress()).willReturn(EIP_1014_ADDRESS);
+        given(frame.getWorldUpdater()).willReturn(worldUpdater);
+        given(contract.getMutable()).willReturn(mutableContract);
+        given(mutableContract.getCode()).willReturn(Bytes.EMPTY);
+        given(worldUpdater.getOrCreate(EIP_1014_ADDRESS)).willReturn(contract);
+        given(frame.getValue()).willReturn(WEI_VALUE);
+        final var maybeReasonToHalt = Optional.of(ExceptionalHaltReason.ILLEGAL_STATE_CHANGE);
+        given(worldUpdater.tryTransferFromContract(
+                        NON_SYSTEM_LONG_ZERO_ADDRESS, EIP_1014_ADDRESS, WEI_VALUE.toLong(), false))
+                .willReturn(maybeReasonToHalt);
+
+        subject.start(frame, tracer);
+
+        verify(frame).setExceptionalHaltReason(maybeReasonToHalt);
+        verify(frame).setState(MessageFrame.State.EXCEPTIONAL_HALT);
+        verify(tracer, never()).traceAccountCreationResult(frame, maybeReasonToHalt);
+    }
+
+    @Test
+    void haltsWithInsufficientGasIfContractExistsWithNonce() {
+        given(frame.getContractAddress()).willReturn(EIP_1014_ADDRESS);
+        given(frame.getWorldUpdater()).willReturn(worldUpdater);
+        given(contract.getMutable()).willReturn(mutableContract);
+        given(mutableContract.getNonce()).willReturn(1L);
+        given(worldUpdater.getOrCreate(EIP_1014_ADDRESS)).willReturn(contract);
+
+        subject.start(frame, tracer);
+
+        verify(worldUpdater, never())
+                .tryTransferFromContract(NON_SYSTEM_LONG_ZERO_ADDRESS, EIP_1014_ADDRESS, WEI_VALUE.toLong(), false);
+        verify(mutableContract, never()).setNonce(INITIAL_CONTRACT_NONCE);
+        final var maybeReasonToHalt = Optional.of(ExceptionalHaltReason.INSUFFICIENT_GAS);
+        verify(frame).setExceptionalHaltReason(maybeReasonToHalt);
+        verify(frame).setState(MessageFrame.State.EXCEPTIONAL_HALT);
+        verify(tracer).traceAccountCreationResult(frame, maybeReasonToHalt);
+    }
+
+    @Test
+    void haltsWithInsufficientGasIfContractExistsWithNonEmptyCode() {
+        given(frame.getContractAddress()).willReturn(EIP_1014_ADDRESS);
+        given(frame.getWorldUpdater()).willReturn(worldUpdater);
+        given(contract.getMutable()).willReturn(mutableContract);
+        given(mutableContract.getCode()).willReturn(CONTRACT_CODE.getBytes());
```

Comment:
mockito question:  why doesn't this test need a getNonce() -> 0 given?  is this a permissive vs strict thing?

Reply:
Mockito returns `0` from any mocked method that returns a numeric type; so you don't have to configure that explicitly.